packages/@react-aria/interactions/src/useMove.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {HTMLAttributes, useMemo, useRef} from 'react';
+
+export interface BaseMoveEvent {
+  pointerType: 'mouse' | 'pen' | 'touch' | 'keyboard'
+}
+
+export interface MoveStartEvent extends BaseMoveEvent{
+  type: 'movestart'
+}
+
+export interface MoveMoveEvent extends BaseMoveEvent{
+  type: 'move',
+  deltaX: number,
+  deltaY: number
+}
+
+export interface MoveEndEvent extends BaseMoveEvent{
+  type: 'moveend'
+}
+
+export type MoveEvent = MoveStartEvent | MoveMoveEvent | MoveEndEvent;
+
+export interface MoveProps {
+  onMoveStart?: (e: MoveStartEvent) => void,
+  onMove: (e: MoveMoveEvent) => void,
+  onMoveEnd?: (e: MoveEndEvent) => void
+}
+
+// const currentTargets: Set<HTMLElement> = new Set();
+let currentTarget: HTMLElement | null = null;
```

Comment:
This is to make sure that only the the deepest element in the tree is moving, not any potential parent using useMove. But adding/deleting `e.target` from a Set doesn't work because the start & end target could be completely different...

Reply:
> What do you mean by start/end target?

For example: onmousedown happens on the slider thumb but onmouseup on the body itself because you moved the cursor off the slider.

> Should it be a map by pointer id?

what about touch and mouse events?

Reply:
> What happens if multiple elements are being moved at once (e.g. multitouch)?

Yeah, that's the problem at the moment. (Moving multiple sliders at the same time should work but the useMove on the slider track shouldn't fire together with the useMove on the thumb)